lib/pushgateway.js

```diff
@@ -79,8 +81,23 @@ async function useGateway(method, job, groupings) {
 			this.registry
 				.metrics()
 				.then(metrics => {
-					req.write(metrics);
-					req.end();
+					if (
+						options.headers &&
+						options.headers['Content-Encoding'] === 'gzip'
+					) {
+						Readable.from([metrics])
```

Comment:
I don't think you need to use `Readable.from` here since it's just one string. That will be faster and avoid the node 12 requirement issue.

Reply:
What is the alternative? Using gzipSync will keep the output buffer in memory.

Reply:
Just [`zlib.gzip(metrics, cb)`](https://nodejs.org/dist/latest-v16.x/docs/api/zlib.html#zlibgzipbuffer-options-callback), optionally put in a Promise if you want.

Reply:
we already have the entire string in memory, does passing it to `gzipSync` cause any real issue?

Reply:
Probably it would be fine simply using `req.write(gzipSync(metrics))`.